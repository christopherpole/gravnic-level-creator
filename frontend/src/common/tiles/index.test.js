import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ENTITIES } from 'gravnic-game';

import { Tile } from './index';

configure({ adapter: new Adapter() });

describe('<Tile />', () => {
  let props;

  beforeEach(() => {
    props = {
      entity: {
        entityId: ENTITIES.FLOOR.id,
        stuck: false,
        shrinking: false,
        fading: false,
        moveSpeed: 100,
      },
    };
  });

  it('Renders without exploding', () => {
    const tile = shallow(<Tile {...props} />);

    expect(tile).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const tile = shallow(<Tile {...props} />);
    expect(toJson(tile)).toMatchSnapshot();
  });

  it('Matches the current snapshot with a mismatching tile id', () => {
    const tile = shallow(<Tile entity={{ ...props.entity, entityId: 'notanactualid' }} />);
    expect(toJson(tile)).toMatchSnapshot();
  });

  it('Matches the current snapshot with an undefined file', () => {
    const tile = shallow(<Tile entity={{ ...props.entity, entityId: undefined }} />);
    expect(toJson(tile)).toMatchSnapshot();
  });
});
