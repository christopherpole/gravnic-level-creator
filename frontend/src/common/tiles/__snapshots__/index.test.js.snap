// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> Matches the current snapshot 1`] = `
<styled.div
  fading={false}
  moveSpeed={0}
  shrinking={false}
  stuck={false}
>
  <FloorTile
    entityId="FLOOR"
    fading={false}
    moveSpeed={0}
    shrinking={false}
    stuck={false}
  />
</styled.div>
`;

exports[`<Tile /> Matches the current snapshot with a mismatching tile id 1`] = `
<styled.div
  fading={false}
  moveSpeed={0}
  shrinking={false}
  stuck={false}
>
  <MissingTile
    entityId="notanactualid"
    fading={false}
    moveSpeed={0}
    shrinking={false}
    stuck={false}
  />
</styled.div>
`;

exports[`<Tile /> Matches the current snapshot with an undefined file 1`] = `
<styled.div
  fading={false}
  moveSpeed={0}
  shrinking={false}
  stuck={false}
>
  <MissingTile
    fading={false}
    moveSpeed={0}
    shrinking={false}
    stuck={false}
  />
</styled.div>
`;
